### PR TITLE
Fix early check mechanism for batch transactions (batch transactions tracker)

### PR DESCRIPTION
**Pull request by CiprianDraghici**
### Issue/Feature
Check the previous signed session, not the last one in a batch
### Reproduce
Issue exists on version `2.` of sdk-dapp.

### Fix
Check only batches in pending
### Contains breaking changes
[x] No

[] Yes

### Updated CHANGELOG
[x] Yes

### Testing
[x] User testing
[] Unit tests

### CODE REVIEW

File: src/hooks/transactions/batch/tracker/useBatchTransactionsTracker.ts [hunk 1]
```diff
@@ -38,14 +40,16 @@ export const useBatchTransactionsTracker = ({
 
   const checkAllBatches = async () => {
     for (const [sessionId, session] of signedTransactionsArray) {
-      if (
-        session.status !== TransactionBatchStatusesEnum.sent &&
-        session.status !== TransactionBatchStatusesEnum.signed
-      ) {
+      const batchId = `${sessionId}-${address}`;
```
Comment (arhtudormorar) [inline src/hooks/transactions/batch/tracker/useBatchTransactionsTracker.ts:43]: best to use a constructor function here

File: src/hooks/transactions/batch/tracker/useBatchTransactionsTracker.ts [hunk 2]
```diff
@@ -38,14 +40,16 @@ export const useBatchTransactionsTracker = ({
...
+
+      const batchDoesNotExists = !Boolean(batches[batchId]);
+      const isPending = session.status !== TransactionBatchStatusesEnum.sent; //&&
```
Comment (arhtudormorar) [inline src/hooks/transactions/batch/tracker/useBatchTransactionsTracker.ts:46]: //&&

File: src/services/transactions/clearTransactions.ts [hunk 3]
```diff
@@ -12,6 +13,13 @@ export function removeTransactionsToSign(sessionId: string) {
 }
 export function removeSignedTransaction(sessionId: string) {
   store.dispatch(clearSignedTransaction(sessionId));
+
+  const account = accountSelector(store.getState());
+  store.dispatch(
+    clearBatchTransactions({
+      batchId: `${sessionId}-${account?.address}`
```
Comment (arhtudormorar) [inline src/services/transactions/clearTransactions.ts:20]: use constructor everywhere for this logic

File: src/hooks/transactions/batch/tracker/useBatchTransactionsTracker.ts (hunk 1 above)
Comment (CiprianDraghici) [reply src/hooks/transactions/batch/tracker/useBatchTransactionsTracker.ts:43]: I don't see the necessity of having a dedicated function for a string concatenation

**Review by arhtudormorar — APPROVED** [review]

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**